src/stretchy.js

```diff
@@ -52,130 +52,512 @@ const mathMLnode = function(label) {
+
+    xcancel: `<line x1='0' y1='0' x2='100%' y2='100%' stroke-width='1.5px'
+stroke='currentcolor'/><line x1='0' y1='100%' x2='100%' y2='0'
+stroke-width='1.5px' stroke='currentcolor'/>`,
 };
 
 const svgSpan = function(group, options) {
-    // Create a span with class(es) that refer to the background-image
-    // and/or the mask-image.
+    // Create a span with inline SVG for the element.
     const label = group.value.label.substr(1);
     let height = 0;
     let depth = 0;
-    const classArray = ["stretchy"];
-    let fileName = "";
+    let imageName = "";
+    let minWidth = 0;
 
     if (utils.contains(["widehat", "widetilde", "undertilde"], label)) {
         // There are four SVG images available for each function.
         // Choose a taller image when there are more characters.
         const numChars = group.value.value.length;
         if (numChars > 5) {
             height = 0.312;
-            fileName = (label === "widehat" ? "widehat" : "tilde") + "4";
+            imageName = (label === "widehat" ? "widehat" : "tilde") + "4";
         } else {
             const imgIndex = [1, 1, 2, 2, 3, 3][numChars];
             if (label === "widehat") {
                 height = [0, 0.24, 0.30, 0.30, 0.36, 0.36][numChars];
-                fileName = "widehat" + imgIndex;
+                imageName = "widehat" + imgIndex;
             } else {
                 height = [0, 0.26, 0.30, 0.30, 0.34, 0.34][numChars];
-                fileName = "tilde" + imgIndex;
+                imageName = "tilde" + imgIndex;
             }
         }
     } else {
         const imgData = katexImagesData[label];
         height = imgData[0];
         depth = imgData[1];
-        fileName = imgData[2];
-        if (label.substr(0, 1) === "x") {
-            classArray.push("x-arrow");     // Lengthen the arrow via padding.
-        }
+        imageName = imgData[2];
+        minWidth = imgData[3];
     }
 
-    let node;
-    if (options.color) {
-        classArray.push(fileName);         // Set span height and IE image.
-        // The next two lines each add a class that CSS will apply
-        // only to browsers that support CSS mask.
-        // IE will not recognize that CSS, so it will fall back to
-        // the background-image set in the previous line of code.
-        classArray.push("mask");             // Over-ride image.
-        classArray.push(fileName + "-mask"); // Set mask-image.
-        node = buildCommon.makeSpan(classArray, [], options);
-        node.style.backgroundColor = options.getColor();
-    } else {
-        classArray.push(fileName);             // Set image and span height.
-        node = buildCommon.makeSpan(classArray, [], options);
+    const span = buildCommon.makeSpan([], [], options);
+    span.height = height;
+    span.depth = depth;
+    const totalHeight = height + depth;
+    span.style.height = totalHeight + "em";
+    if (minWidth > 0) {
+        span.style.minWidth = minWidth + "em";
```

Comment:
Exception: minWidth for the braces is set large enough to avoid a discontinuity where sloping parts of the brace match up.